ksqldb-engine/src/main/java/io/confluent/ksql/util/SandboxedSharedKafkaStreamsRuntimeImpl.java

```diff
@@ -74,15 +75,30 @@ public void register(
     log.info("Registering query {} for validation for runtime {}", queryId, getApplicationId());
     collocatedQueries.put(queryId, binpackedPersistentQueryMetadata);
     try {
-      if (!kafkaStreams.getTopologyByName(queryId.toString()).isPresent()) {
-        kafkaStreams.addNamedTopology(binpackedPersistentQueryMetadata.getTopologyCopy(this));
+      if (!kafkaStreams.getTopologyByName(
+          binpackedPersistentQueryMetadata.getQueryId().toString()).isPresent()) {
```

Comment:
I think the reason we passed in the second param `queryId` is that, except for the non-testing caller, `start()` which just pass in itself and the `queryId` field, some testing code used a mock and hence cannot get `binpackedPersistentQueryMetadata.getQueryId()` from the mock, right? If not, then we do not need this second param in `register`, but if yes, then it seems we should not replace `queryId` to `binpackedPersistentQueryMetadata.getQueryId()`.

Reply:
I don't remember why we took the query ID as a parameter. I will try removing it